validator/client/src/test/java/tech/pegasys/teku/validator/client/loader/KeystoreLockerTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client.loader;
+
+import static java.nio.file.StandardOpenOption.CREATE_NEW;
+import static org.assertj.core.api.Assertions.assertThat;
+import static tech.pegasys.teku.validator.client.loader.KeystoreLocker.longPidToNativeByteArray;
+import static tech.pegasys.teku.validator.client.loader.KeystoreLocker.nativeByteArrayToLong;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.io.TempDir;
+import tech.pegasys.teku.util.config.InvalidConfigurationException;
+
+public class KeystoreLockerTest {
+
+  private final KeystoreLocker keystoreLocker = new KeystoreLocker();
+
+  @Test
+  void shouldLockKeystoreFileAndFailWhenTryingCreateLockForLockedFile(
+      final @TempDir Path keystoreFile) throws Exception {
+    Assertions.assertThatCode(() -> keystoreLocker.lockKeystore(keystoreFile))
+        .doesNotThrowAnyException();
+    Assertions.assertThatThrownBy(() -> keystoreLocker.lockKeystore(keystoreFile))
+        .isExactlyInstanceOf(InvalidConfigurationException.class);
+  }
+
+  @Test
+  @EnabledOnOs({OS.LINUX, OS.MAC})
+  void deleteLockfileIfTheProcessIsNotAlive(final @TempDir Path keystoreFile) throws Exception {
+    Process process = new ProcessBuilder("/bin/sleep", "1").start();
+    Thread.sleep(1500);
+    long pid = process.pid();
+    assertThat(process.isAlive()).isFalse();
```

Comment:
Rather than running an external program and sleeping, I'd suggest just using a really high value (probably `Long.MAX_VALUE`). Linux defaults to a max PID of 32768 and can only go up to 2^22 so it should be safe enough even if other OS's vary.

Reply:
Makes sense. Done.